src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -713,11 +713,23 @@ private static void CheckRestrictedTypeReceiver(BoundExpression expression, Comp
 
             if (!result.Succeeded)
             {
-                // If the arguments had an error reported about them then suppress further error
-                // reporting for overload resolution. 
-
-                if (!analyzedArguments.HasErrors)
+                if (analyzedArguments.HasErrors)
+                {
+                    // Errors for arguments have already been reported, except for unbound lambdas.
```

Comment:
Would it make sense to report errors for unbound lambdas where it is done for other arguments? Also since we are binding the lambdas for error recovery, would it make sense to capture BoundLambda nodes as the arguments in the tree so that SemanticModel wouldn't have to deal with UnboundLambdas again?